File: tools/scratch_check_adrs.py
import os
import re
import sys

adr_dir = "/Users/huihui/Documents/qijiu-desktop-pet/docs/decisions"
adrs = sorted([f for f in os.listdir(adr_dir) if f.startswith("ADR-") and f.endswith(".md")])

expected_headers = ["Status", "Date", "Context", "Decision", "Alternatives Considered", "Consequences"]

issues = []

for adr in adrs:
    path = os.path.join(adr_dir, adr)
    with open(path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    headers = re.findall(r'^##\s+(.*)', content, re.MULTILINE)
    headers_clean = [h.strip() for h in headers]
    
    missing = [h for h in expected_headers if not any(h.lower() in existing.lower() for existing in headers_clean)]
    
    if missing:
        issues.append(f"{adr} missing: {missing}")
    
    status_match = re.search(r'^##\s+Status.*?\n(.*?)(?=\n##)', content, re.DOTALL | re.IGNORECASE)
    if status_match:
        status_val = status_match.group(1).strip()
        if status_val not in ["Accepted", "Superseded", "Deprecated"]:
            issues.append(f"{adr} status format wrong: {repr(status_val[:30])}")
    else:
        issues.append(f"{adr} missing Status value block")

for i in issues:
    print(i)

File: tools/scratch_check_adrs.js
const fs = require('fs');
const path = require('path');

const adrDir = '/Users/huihui/Documents/qijiu-desktop-pet/docs/decisions';
const adrs = fs.readdirSync(adrDir).filter(f => f.startsWith('ADR-') && f.endsWith('.md')).sort();

const expectedHeaders = ['Status', 'Date', 'Context', 'Decision', 'Alternatives Considered', 'Consequences'];
const issues = [];

adrs.forEach(adr => {
  const content = fs.readFileSync(path.join(adrDir, adr), 'utf-8');
  const headers = [];
  const regex = /^##\s+(.*)/gm;
  let match;
  while ((match = regex.exec(content)) !== null) {
    headers.push(match[1].trim());
  }

  const missing = expectedHeaders.filter(h => !headers.some(existing => existing.toLowerCase().includes(h.toLowerCase())));
  if (missing.length > 0) {
    issues.push(`${adr} missing: ${missing.join(', ')}`);
  }

  const statusRegex = /^##\s+Status.*?\n(.*?)(?=\n##)/ims;
  const statusMatch = statusRegex.exec(content);
  if (statusMatch) {
    const val = statusMatch[1].trim();
    if (!['Accepted', 'Superseded', 'Deprecated'].includes(val)) {
      issues.push(`${adr} status format wrong: ${val.substring(0, 30).replace(/\n/g, ' ')}`);
    }
  } else {
    issues.push(`${adr} missing Status value block`);
  }
});

issues.forEach(i => console.log(i));
